draft-irtf-cfrg-hpke.md

```diff
@@ -703,14 +703,15 @@ def OpenAuthPSK(enc, skR, info, aad, ct, psk, pskID, pkS):
 | 0x0020 | DHKEM(Curve25519) | 32   | 32  | {{?RFC7748}}   |
 | 0x0021 | DHKEM(Curve448)   | 56   | 56  | {{?RFC7748}}   |
 
-For the NIST curves P-256 and P-521, the Marshal function of the DH
-scheme produces the normal (non-compressed) representation of the
-public key, according to {{SECG}}.  When these curves are used, the
-recipient of an HPKE ciphertext MUST validate that the ephemeral public
-key `pkE` is on the curve and part of the correct prime-order subgroup.
-For authenticated modes the same validation MUST be done for the static
-public key `pkS`.  The relevant full public-key validation procedure is
-defined in {{keyagreement}}.
+For the NIST curves P-256, P-384 and P-521, the Marshal function of the
+DH scheme produces the normal (non-compressed) representation of the
+public key, according to {{SECG}}.  When these curves are used, senders
+and recipients MUST perform a full public-key validation as defined
+in {{keyagreement}}, which includes validating that a public key is on
+the curve and part of the correct prime-order subgroup.  The sender
+MUST validate the recipient's public key `pkR`.  The recipient MUST
+validate the ephemeral public key `pkE`.  In authenticated modes, the
+recipient MUST validate the sender's static public key `pkS`.
```

Comment:
Can we move the text from "The sender MUST..." to the end of this paragraph to its own paragraph? By folding this into the paragraph specific to NIST curves, it might read as though it only applies to NIST curves, which is not the intent.

Reply:
I am not sure how to deal with this, so I'll lay out what my current view is: In the current version of HPKE, there is only NIST curves and Curve25519/448. Public key validation applies for the NIST curves but not for Curve25519/448. Which means in this context, it applies only to NIST curves.

Reply:
This seems to be conflating two things: what curves require validation, and what keys are validated. I think we should separate these two. (I also think we should validate curve25519 and curve448 points, too, as the previous text suggested.)